.ado/xcconfig/publish_overrides.xcconfig

```diff
@@ -0,0 +1,13 @@
+// For publish builds, only provide line tables for symbolizing crashes
+CLANG_DEBUG_INFORMATION_LEVEL=line-tables-only
+OTHER_SWIFT_FLAGS=-gline-tables-only
+
+// Optimize for size in publish builds
+SWIFT_OPTIMIZATION_LEVEL=-Osize
+SWIFT_REFLECTION_METADATA_LEVEL=none
```

Comment:
Don't we also publish debug builds?

Reply:
Oh, yes. I assume them that lines 6 and 7 have to be release only then?

Reply:
I had to not use these as the xcodebuild invocation would then fail. I'm not sure why, but I'll probably need to remove them from the review and add back later, say, for ARM64 support on macOS

Reply:
Ended up not using any of the xcconfig overrides. Will revisit in a future change, but will leave it off for now

Reply:
But we still need to ensure those optimizations are on for release builds (shouldn't be in xcconfig overrides but in static xcconfigs that live with the project itself and not just as part of the publish pipelines

Reply:
As it turns out, we already have some of these settings in xcconfigs we get for free from `react-native-test-app`
https://github.com/microsoft/react-native-test-app/pull/151/files
https://github.com/microsoft/react-native-test-app/blob/master/macos/ReactTestApp/ReactTestApp.common.xcconfig
https://github.com/microsoft/react-native-test-app/blob/master/macos/ReactTestApp/ReactTestApp.debug.xcconfig
https://github.com/microsoft/react-native-test-app/blob/master/macos/ReactTestApp/ReactTestApp.release.xcconfig
https://github.com/microsoft/react-native-test-app/blob/master/ios/ReactTestApp/ReactTestApp.common.xcconfig
https://github.com/microsoft/react-native-test-app/blob/master/ios/ReactTestApp/ReactTestApp.debug.xcconfig
https://github.com/microsoft/react-native-test-app/blob/master/ios/ReactTestApp/ReactTestApp.release.xcconfig